### PR TITLE
vendor: bump github.com/evanphx/json-patch

### PR DESCRIPTION
**Pull request by CaoShuFeng**
Grab important bug fix that can cause a `panic()` from this package on
certain inputs. See https://github.com/evanphx/json-patch/pull/64

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by CaoShuFeng** [comment]
cc @felixfbecker

**Comment by caesarxuchao** [comment]
/assign @cblecker 

This is fixing a similar bug in json patch as in https://github.com/kubernetes/kubernetes/pull/68428. It's good to have it in 1.12, but I'm not a milestone modifier.

**Comment by liggitt** [comment]
I'm out of the office this week, assigning sig-api-machinery leads for review
/assign @deads2k @lavalamp

**Comment by cblecker** [comment]
/milestone v1.12
Adding to milestone to track this.

**Comment by lavalamp** [comment]
With a bit of investigation, I think we can't accept this change. It adds negative indexing (counting backwards) for array indices, which is not how the specification claims the index handling should work: https://tools.ietf.org/html/rfc6901#section-4

(See JSON Patch spec claiming it uses JSON Pointer notation for the path: https://tools.ietf.org/html/rfc6902#section-4)

(Please let me know if I am missing some other part of the spec, I'm not an expert on JSON patch I just know when things look weird and can use "find" on the spec document.)

We can't start allowing non-standard requests or we'd likely have to support them forever. Hopefully this library doesn't already allow other non standard notation :(

**Comment by lavalamp** [comment]
Since this is clearly a bug fix I don't think there's any urgency to get it into the .0 release. We can always pick it up in a patch.

File: vendor/github.com/evanphx/json-patch/patch.go [hunk 1]
```diff
@@ -389,17 +389,13 @@ func (d *partialArray) add(key string, val *lazyNode) error {
 
 	cur := *d
 
-	if idx < 0 {
-		idx *= -1
```
Comment (liggitt) [inline vendor/github.com/evanphx/json-patch/patch.go]: didn't this already do negative indexing?

Comment (lavalamp) [reply vendor/github.com/evanphx/json-patch/patch.go]: SIGH

**Comment by lavalamp** [comment]
I filed https://github.com/evanphx/json-patch/issues/65

**Comment by liggitt** [comment]
needs priority label to merge for 1.12

**Pull request merged**